samcli/commands/validate/lib/sam_template_validator.py

```diff
@@ -139,6 +141,32 @@ def _replace_local_image(self):
                 if "ImageUri" not in properties:
                     properties["ImageUri"] = "111111111111.dkr.ecr.region.amazonaws.com/repository"
 
+    def _replace_local_openapi(self):
+        """
+        Applies Transform Include of DefinitionBody.
+        Without this validation fails.
+        """
+        # look for transform that includes a yaml definition like this:
+        # DefinitionBody:
+        #   'Fn::Transform':
+        #     Name: AWS::Include
+        #     Parameters:
+        #       Location: openapi.yaml
+        resources = self.sam_template.get("Resources", {})
+        for _, resource in resources.items():
+            if not resource.get("Type", "") in [AWS_SERVERLESS_API, AWS_SERVERLESS_HTTPAPI]:
+                continue
+
+            transform_dict = resource.get("Properties", {}).get("DefinitionBody", {}).get("Fn::Transform", {})
+            if transform_dict.get("Name", "") != "AWS::Include":
+                continue
+
+            location_prop = transform_dict.get("Parameters", {}).get("Location", "")
+            if not os.path.isfile(location_prop):
```

Comment:
Can we use `pathlib` instead of `os.path`?

https://docs.python.org/3/library/pathlib.html#pathlib.Path.is_file would be the function call.

Reply:
Cheers. Have changed to use pathlib.